pkg/controller/operator.go

```diff
@@ -105,15 +104,23 @@ func (s *Operator) Run(ctx context.Context, controller *controllercmd.Controller
 	// the last sync time, if any was set
 	statusReporter := status.NewController(configClient, gatherKubeClient.CoreV1(), configObserver, os.Getenv("POD_NAMESPACE"))
 
+	// anonymizer is responsible for anonymizing sensitive data, it can be configured to disable specific anonymization
+	anonymizer, err := anonymization.NewAnonymizerFromConfigClient(ctx, configObserver, kubeClient, configClient)
+	if err != nil {
+		return err
```

Comment:
Hmm this can lead to failing IO (e.g it won't be able to read the `cluster-config-v1` CM -> fails horribly) even when the obfuscation is disabled right? I think we want to avoid such a situation. I guess we can check the new flag right here and create the anonymizer only when it's enabled. WDYT?

Reply:
+1